taxi/tests.py

```diff
@@ -1,3 +1,168 @@
+            list(Car.objects.filter(manufacturer__name="manufacturer")),
+            []
+        )
+
+
+class TestAdminRequired(TestCase):
+    def setUp(self) -> None:
+        self.admin_user = get_user_model().objects.create_superuser(
+            username="admin",
+            password="admin"
+        )
+
+    def test_without_login(self):
+        url = reverse("taxi:index")
+        res = self.client.get(url)
+        self.assertEqual(res.status_code, 302)
+
+    def with_login(self):
+        self.client.force_login(self.admin_user)
+        url = reverse("taxi:index")
+        res = self.client.get(url)
+        self.assertEqual(res.status_code, 200)
+
+
+class TestViews(TestCase):
+    def setUp(self) -> None:
+        self.admin_user = get_user_model().objects.create_superuser(
+            username="admin",
+            password="admin"
+        )
+        self.client = Client()
+        self.manufacturer = Manufacturer.objects.create(
+            name="manufacturer",
+            country="country"
+        )
+
+    def test_view_index_count(self) -> None:
+        self.client.force_login(self.admin_user)
+        url = reverse("taxi:index")
+        res = self.client.get(url)
+        self.assertEqual(res.context["num_drivers"], 1)
+        self.assertEqual(res.context["num_cars"], 0)
+        self.assertEqual(res.context["num_manufacturers"], 1)
+
+
+class FormTests(TestCase):
+    def setUp(self) -> None:
+        self.driver = get_user_model().objects.create_superuser(
+            username="admin",
+            password="admin"
+        )
+        self.manufacturer = Manufacturer.objects.create(
+            name="manufacturer",
+            country="country"
+        )
+        self.car = Car.objects.create(
+            model="model",
+            manufacturer=self.manufacturer,
+        )
+        self.car.drivers.add(self.driver)
+
+    def test_car_form_is_valid(self):
+        form_data = {
+            "model": "test",
+            "manufacturer": self.manufacturer.id,
+            "drivers": [self.driver.id]
+        }
+        form = CarForm(data=form_data)
+        self.assertEqual(form.is_valid(), True)
+
+    def test_car_form_is_not_valid(self):
+        form_data = {
+            "manufacturer": self.manufacturer.id,
+            "drivers": [self.driver.id]
+        }
+        form = CarForm(data=form_data)
+        self.assertEqual(form.is_valid(), False)
+
+        form_data = {
+            "model": "test",
+            "drivers": [self.driver.id]
+        }
+        form = CarForm(data=form_data)
+        self.assertEqual(form.is_valid(), False)
+
+        form_data = {
+            "model": "test",
+            "manufacturer": self.manufacturer.id,
+        }
+        form = CarForm(data=form_data)
+        self.assertEqual(form.is_valid(), False)
+
+        form_data = {
+            "model": "test",
+            "manufacturer": self.manufacturer.id,
+            "drivers": self.driver.id
```

Comment:
The `drivers` field should be a list of IDs, not a single ID. Change `self.driver.id` to `[self.driver.id]` to match the expected data type for the `drivers` field in the `CarForm`.